File: csharp/ToolGood.Algorithm/Operands/OperandKeyValue.cs
using ToolGood.Algorithm.Enums;

namespace ToolGood.Algorithm.Operands
{
	internal sealed class OperandKeyValue : Operand
	{
		private readonly KeyValue _value;
		public OperandKeyValue(KeyValue obj)
		{
			_value = obj;
		}
		public override bool IsArrayJson => true;
		public override OperandType Type => OperandType.ARRAYJSON;
		public KeyValue Value => _value;
	}
}

File: java/src/main/java/toolgood/algorithm/operands/OperandKeyValue.java
package toolgood.algorithm.operands;

import toolgood.algorithm.Operand;
import toolgood.algorithm.enums.OperandType;

public final class OperandKeyValue extends Operand {
    private final KeyValue _value;

    public OperandKeyValue(KeyValue obj) {
        _value = obj;
    }

    @Override
    public boolean IsArrayJson() { return true; }

    @Override
    public OperandType Type() { return OperandType.ARRAYJSON; }

    public KeyValue Value() { return _value; }
}
